data/scenarios/Challenges/Sokoban/Gadgets/no-reverse.yaml

```diff
@@ -0,0 +1,81 @@
+version: 1
+name: No-reverse gadget
+author: Karl Ostmo
+description: |
+  Once a robot passes from left to right, they cannot return they way they came.
+creative: false
+seed: 0
+objectives:
+  - goal:
+      - Grab the flower.
+    condition: |
+      as base {
+        has "flower";
+      };
+robots:
+  - name: base
+    dir: [0, -1]
+    display:
+      attr: gold
+    devices:
+      - branch predictor
```

Comment:
In general, for all these sokoban scenarios, can we also equip a `compass` and an `ADT calculator`?  The way I like to implement "driving mode" makes use of them.  I don't think those devices fundamentally alter the nature of the challenge.